File: Programming Skills II/150. Evaluate Reverse Polish Notation/150. Evaluate Reverse Polish Notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    let stack = [];
    let t = 0, t1 = 0;
    for( i in tokens ){
        if(tokens[i] == "+"){
            t = stack.pop();
            t1 = stack.pop();
            stack.push(t+t1);
        }else if(tokens[i] == "-"){
            t = stack.pop();
            t1 = stack.pop();
            stack.push(t1-t);
        }else if(tokens[i] == "*"){
            t = stack.pop();
            t1 = stack.pop();
            stack.push(t*t1);
        }else if(tokens[i] == "/"){
            t = stack.pop();
            t1 = stack.pop();
            stack.push( parseInt((t1/t), 10) );
        }else{
            stack.push(parseInt(tokens[i]));
        }
    }
    return stack.pop();
};

File: Programming Skills II/150. Evaluate Reverse Polish Notation/150. Evaluate Reverse Polish Notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> st = new Stack();
        int t, t2;
        for(int i=0; i<tokens.length; ++i){
            if(tokens[i].equals("+")){
                t = st.pop();
                t2 = st.pop();
                st.push(t+t2);
            }else if(tokens[i].equals("-")){
                t = st.pop();
                t2 = st.pop();
                st.push(t2-t);
            }else if(tokens[i].equals("*")){
                t = st.pop();
                t2 = st.pop();
                st.push(t2*t);
            }else if(tokens[i].equals("/")){
                t = st.pop();
                t2 = st.pop();
                st.push(t2/t);
            }else{
                st.push( Integer.parseInt((tokens[i])) );
            }
        }
        return st.pop();
    }
}
